core/src/main/java/org/infinispan/security/MutableRolePermissionMapper.java

```diff
@@ -0,0 +1,12 @@
+package org.infinispan.security;
+
+import java.util.concurrent.CompletionStage;
+
+/**
+ * @since 14.0
+ **/
+public interface MutableRolePermissionMapper extends RolePermissionMapper {
+   CompletionStage<Void> addRole(Role role);
```

Comment:
didn't notice this before, but if this is a public interface, it would be better to have Javadoc.